### PR TITLE
Sensors disconnect detection

### PR DESCRIPTION
**Pull request by HilelAmiel**
- check command 
- subsystem error code check

### CODE REVIEW

File: src/main/java/frc/robot/components/Pigeon.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public void reset() {
         setYaw(0);
     }
 
+    public ErrorCode resetGyroWithErrorCode() {
+        return setYaw(0);
+    }
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/components/Pigeon.java:50]: we can just make reset() return ErrorCode. I don't see why would we need two separate pigeon reset methods

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 2]
```diff
@@ -37,7 +42,11 @@ public Shooter() {
         rightTalonFX.setNeutralMode(NeutralMode.Coast);
         rightTalonFX.configClosedloopRamp(robotConstants.shooterConstants.kRampTime);
         rightTalonFX.configOpenloopRamp(robotConstants.shooterConstants.kRampTime);
-        rightTalonFX.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
+        
+        if(rightTalonFX.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0) 
+            != ErrorCode.OK)
+            DriverStationLogger.logErrorToDS("Could not set right shooter encoder");
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java]: Use the new DriverStationLogger method instead

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 3]
```diff
@@ -28,7 +29,11 @@ public Shooter() {
         leftTalonFX.setNeutralMode(NeutralMode.Coast);
         leftTalonFX.configClosedloopRamp(robotConstants.shooterConstants.kRampTime);
         leftTalonFX.configOpenloopRamp(robotConstants.shooterConstants.kRampTime);
-        leftTalonFX.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
+
+        if(leftTalonFX.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0) 
+            != ErrorCode.OK)
+            DriverStationLogger.logErrorToDS("Could not set left shooter encoder");
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java]: Use the new DriverStationLogger method instead

File: src/main/java/frc/robot/subsystems/loader/Loader.java [hunk 4]
```diff
@@ -17,11 +20,20 @@
...
+        talonSRX.configOpenloopRamp(robotConstants.loaderConstants.kRampRate);
         talonSRX.setNeutralMode(NeutralMode.Coast);
+        
         talonSRX.configSupplyCurrentLimit(
             new SupplyCurrentLimitConfiguration(true, robotConstants.loaderConstants.kCurrentLimit,
                 robotConstants.loaderConstants.kThresholdLimit, robotConstants.loaderConstants.kTimeout));
+        
+        if(talonSRX.configSelectedFeedbackSensor(TalonSRXFeedbackDevice.CTRE_MagEncoder_Relative, 0, 0) 
+            != ErrorCode.OK)
+            DriverStationLogger.logErrorToDS("Could not set loader encoder");
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/loader/Loader.java]: Use the new DriverStationLogger method instead

File: src/main/java/frc/robot/utils/DriverStationLogger.java [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 import static frc.robot.Robot.led;
 
+import com.ctre.phoenix.ErrorCode;
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/utils/DriverStationLogger.java]: import static should be after import

**Review by EliyaFishman — CHANGES_REQUESTED** [review]
change

**Comment by EliyaFishman** [comment]
[IsAlive sensor detecting](https://trello.com/c/o3t81oRu/124-isalive-sensor-detecting)

File: src/main/java/frc/robot/components/Pigeon.java (hunk 1 above)
Comment (HilelAmiel) [reply src/main/java/frc/robot/components/Pigeon.java:50]: The return type is incompatible with Gyro.reset()

File: src/main/java/frc/robot/commands/SensorCheck.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  @Override
+  public void end(boolean interrupted) {
+    drivetrain.stopMove();
+    shooter.stopMove();
+    loader.stopMove();
+    // Check if none error detected, reports that all encoders are fine 
+    int i = 0;
+    for (; i < hasFoundError.length; i++) {
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/commands/SensorCheck.java:67]: move int i to for

**Review by EliyaFishman — CHANGES_REQUESTED** [review]
Really good but You still didn't change that Pigeon should only have one reset method instead of two so, please change

File: src/main/java/frc/robot/commands/SensorCheck.java (hunk 6 above)
Comment (HilelAmiel) [reply src/main/java/frc/robot/commands/SensorCheck.java:67]: but i's scope would be only in the for block and I use i outside the for block and that is why he should be created locally on the method block

**Review by HilelAmiel — COMMENTED** [review]
lol

**Review by EliyaFishman — APPROVED** [review]

**Pull request merged**